src/model/encoding/convertFromHTMLToContentBlocks.js

```diff
@@ -84,6 +85,14 @@ const inlineTags = {
   u: 'UNDERLINE',
 };
 
+const knownListItemDepthClasses = {
+  [cx('public/DraftStyleDefault/depth0')]: 0,
+  [cx('public/DraftStyleDefault/depth1')]: 1,
+  [cx('public/DraftStyleDefault/depth2')]: 2,
+  [cx('public/DraftStyleDefault/depth3')]: 3,
+  [cx('public/DraftStyleDefault/depth4')]: 4,
+};
```

Comment:
From what I understand, list indentations above 4 would not be preserved. Should this be changed to support arbitrary depth?

I imagine the current approach is based on the class names defined in https://github.com/facebook/draft-js/blob/232791a4e92d94a52c869f853f9869367bdabdac/src/component/contents/DraftEditorContents-core.react.js#L58-L62. An alternative would be to generate those class names dynamically with the right depth suffix in `getListItemClasses`, and here in `getListItemDepth` get the depth by looking for a number right after the `cx('public/DraftStyleDefault/depth')` string.

I realise this is a bit more work but it would be nice to support all use cases of depth. While the `getListItemClasses` code does stop at depth 4 at the moment, this is easily extensible via `blockStyleFn`, whereas the conversion from HTML isn't.

Reply:
Hi @thibaudcolas - 
We are working on replacing this approach of using CSS classes with true nested blocks, in a longer term project. So for now we're not adding improvements to the old CSS code, but ultimately will be able to support arbitrary nesting. That is indeed a good idea. :)

I would not block this diff on that change either way.